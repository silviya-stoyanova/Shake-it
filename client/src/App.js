import React, { Component } from 'react';
import { BrowserRouter } from 'react-router-dom'
import { ToastContainer } from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css';
import './static/css/site.css'
import './static/css/scrollbar.css'

import sessionManager from './components/utilities/session-util'
import observer from './components/utilities/observer'

import MyRoutes from './components/routes/router'
import Header from './components/common/header'
import Footer from './components/common/footer'

const defaultValue = { isLogged: false }
const { Consumer: UserInfoConsumer, Provider: UserInfoProvider } = React.createContext(defaultValue)

class App extends Component {
  constructor(props) {
    super(props)
    this.state = {
      isLogged: sessionManager.isUserLogged(),
      username: sessionManager.getUsername(),
      role: sessionManager.getUserRole()
    }

    observer.subscribe('userLogin', this.onUserLogin)
    observer.subscribe('userLogout', this.onUserLogout)
  }

  onUserLogin = () => {
    this.setState(prevState => ({ // asyncronous
      isLogged: true,
      username: sessionManager.getUsername(),
      role: sessionManager.getUserRole()
    }))
  }

  onUserLogout = () => {
    this.setState(prevState => ({
      isLogged: false,
      username: '',
      role: []
    }))
  }

  render() {
    const { isLogged, username, role } = this.state

    // console.log(this.props.navigation)

    return (
      <UserInfoProvider value={{ isLogged, username, role }}>
        <BrowserRouter>
          <ToastContainer />
          <Header />

          {
            // <div className="main-container">
          }
          <MyRoutes />
          {
            // </div>
          }

          <Footer />
        </BrowserRouter>
      </UserInfoProvider>
    )
  }

  componentDidMount() {
    sessionManager.isUserLogged()
      ? this.setState({ isLogged: true })
      : this.setState({ isLogged: false })
  }
}

export { UserInfoConsumer }
export default App