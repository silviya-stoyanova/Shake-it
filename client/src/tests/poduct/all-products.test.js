import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { mount } from 'enzyme'
import { UserInfoProvider } from '../../App'
import AllProducts from '../../components/product/all-products'

const mockManyProductsFetch = () => {
    const sampleData = JSON.stringify([
        {
            _id: 1,
            title: 'Tova e title',
            image: 'img',
            price: 5,
            likes: ['firstUserId', 'secondUserId', 'thirdUserId']
        },
        {
            _id: 2,
            title: 'Tova e title 2',
            image: 'img-2',
            price: 6,
            likes: ['firstUserId', 'secondUserId']
        },
        {
            _id: 3,
            title: 'Tova e title 3',
            image: 'img-3',
            price: 3.8,
            likes: ['firstUserId']
        }
    ])

    // wrap the data in a new Response object to be able to handle it correctly in withProcessForm.js
    return Promise.resolve(new Response(sampleData))
}

const mockAProductFetch = () => {
    const sampleData = JSON.stringify([
        {
            _id: 1,
            title: 'Tova e title',
            image: 'img',
            price: 5,
            likes: ['firstUserId', 'secondUserId', 'thirdUserId']
        }
    ])

    return Promise.resolve(new Response(sampleData))
}

const mockNoProductsFetch = () => {
    const sampleData = JSON.stringify([])
    return Promise.resolve(new Response(sampleData))
}

describe('tests for the component AllProducts, a.k.a. Home page', () => {

    describe('guest functionalities tests', () => {
        it('should render loading gif for all [guests] until the info is fetched', () => {
            const defaultUserValue = { isLogged: false, username: '', role: '' }

            const wrapper = mount(
                <UserInfoProvider value={defaultUserValue}>
                    <AllProducts />
                </UserInfoProvider>
            )

            expect(wrapper.find('img[alt="loading-img"]').length).toEqual(1)
        })

        it('should render a div saying that there are no products for all [guests]', () => {
            const defaultUserValue = { isLogged: false, username: '', role: '' }

            const wrapper = mount(
                <UserInfoProvider value={defaultUserValue}>
                    <AllProducts service={mockNoProductsFetch} />
                </UserInfoProvider>
            )

            mockNoProductsFetch()
                .then(res => res.json())
                .then(res => {
                    wrapper.setState({
                        products: res,
                        isFetched: true
                    })

                    wrapper.update()

                    expect(wrapper.html()).toMatchSnapshot()

                    expect(wrapper.find('div.no-products-container').length).toEqual(1)
                    expect(wrapper.find('div.no-products-text-container').length).toEqual(1)

                    expect(wrapper.find('h1').length).toEqual(1)
                    expect(wrapper.find('h1').text()).toEqual('Our products are so desired that we could not predict we would run out of stock this soon..')

                    expect(wrapper.find('h2').length).toEqual(1)
                    expect(wrapper.find('h2').text()).toEqual('Sorry!')

                    expect(wrapper.find('div.no-products-img').length).toEqual(1)
                })
        })

        it('should render one product without any links below it for all [guests]', () => {
            const defaultUserValue = { isLogged: false, username: '', role: '' }

            const wrapper = mount(
                <BrowserRouter>
                    <UserInfoProvider value={defaultUserValue}>
                        <AllProducts service={mockAProductFetch} />
                    </UserInfoProvider>
                </BrowserRouter>
            )

            mockAProductFetch()
                .then(res => res.json())
                .then(res => {
                    wrapper.setState({
                        products: res,
                        isFetched: true
                    })
                    wrapper.update()

                    expect(wrapper.html()).toMatchSnapshot()
                    expect(wrapper.find('a[href="/product/details/1"]').length).toEqual(2)

                    expect(wrapper.find('a[href="/product/like/1"]').length).toEqual(0)
                    expect(wrapper.find('a[href="/product/edit/1"]').length).toEqual(0)
                    expect(wrapper.find('a[href="/product/delete/1"]').length).toEqual(0)
                    expect(wrapper.find('a[href="/cart/add/1"]').length).toEqual(0)
                })
        })

        it('should render three products without any links below it for all [guests]', () => {
            const defaultUserValue = { isLogged: false, username: '', role: '' }

            const wrapper = mount(
                <BrowserRouter>
                    <UserInfoProvider value={defaultUserValue}>
                        <AllProducts service={mockManyProductsFetch} />
                    </UserInfoProvider>
                </BrowserRouter>
            )

            mockManyProductsFetch()
                .then(res => res.json())
                .then(res => {
                    wrapper.setState({
                        products: res,
                        isFetched: true
                    })

                    wrapper.update()

                    expect(wrapper.html()).toMatchSnapshot()

                    res.map(product => {
                        expect(wrapper.find(`a[href="/product/details/${product._id}"]`).length).toEqual(2)
                        expect(wrapper.find(`a[href="/product/like/${product._id}"]`).length).toEqual(0)
                        expect(wrapper.find(`a[href="/product/edit/${product._id}"]`).length).toEqual(0)
                        expect(wrapper.find(`a[href="/product/delete/${product._id}"]`).length).toEqual(0)
                        expect(wrapper.find(`a[href="/cart/add/${product._id}"]`).length).toEqual(0)
                    })
                })
        })
    })

    describe('user functionalities tests', () => {
        it('should render loading gif for all [users] until the info is fetched', () => {
            const defaultUserValue = { isLogged: true, username: 'my-name-is-yolo', role: 'User' }

            const wrapper = mount(
                <UserInfoProvider value={defaultUserValue}>
                    <AllProducts />
                </UserInfoProvider>
            )

            expect(wrapper.find('img[alt="loading-img"]').length).toEqual(1)
        })

        it('should render a div saying that there are no products for all [users]', () => {
            const defaultUserValue = { isLogged: false, username: 'cucumber', role: 'User' }

            const wrapper = mount(
                <UserInfoProvider value={defaultUserValue}>
                    <AllProducts service={mockNoProductsFetch} />
                </UserInfoProvider>
            )

            mockNoProductsFetch()
                .then(res => res.json())
                .then(res => {
                    wrapper.setState({
                        products: res,
                        isFetched: true
                    })

                    wrapper.update()

                    expect(wrapper.html()).toMatchSnapshot()

                    expect(wrapper.find('div.no-products-container').length).toEqual(1)
                    expect(wrapper.find('div.no-products-text-container').length).toEqual(1)

                    expect(wrapper.find('h1').length).toEqual(1)
                    expect(wrapper.find('h1').text()).toEqual('Our products are so desired that we could not predict we would run out of stock this soon..')

                    expect(wrapper.find('h2').length).toEqual(1)
                    expect(wrapper.find('h2').text()).toEqual('Sorry!')

                    expect(wrapper.find('div.no-products-img').length).toEqual(1)
                })
        })

        it('should render one product with links for [like] and [add to cart] below for all [users]', () => {
            const defaultUserValue = { isLogged: true, username: 'az-sam-tova', role: 'User' }

            const wrapper = mount(
                <BrowserRouter>
                    <UserInfoProvider value={defaultUserValue}>
                        <AllProducts service={mockAProductFetch} />
                    </UserInfoProvider>
                </BrowserRouter>
            )

            mockAProductFetch()
                .then(res => res.json())
                .then(res => {

                    wrapper.setState({
                        products: res,
                        isFetched: true
                    })
                    wrapper.update()

                    expect(wrapper.html()).toMatchSnapshot()
                    expect(wrapper.find('a[href="/product/details/1"]').length).toEqual(2)

                    expect(wrapper.find('a[href="/product/like/1"]').length).toEqual(1)
                    expect(wrapper.find('a[href="/product/edit/1"]').length).toEqual(0)
                    expect(wrapper.find('a[href="/product/delete/1"]').length).toEqual(0)
                    expect(wrapper.find('a[href="/cart/add/1"]').length).toEqual(1)
                })
        })

        it('should render three products with links for [like] and [add to cart] below for all [users]', () => {
            const defaultUserValue = { isLogged: true, username: 'az-sam-tova', role: 'User' }

            const wrapper = mount(
                <BrowserRouter>
                    <UserInfoProvider value={defaultUserValue}>
                        <AllProducts service={mockManyProductsFetch} />
                    </UserInfoProvider>
                </BrowserRouter>
            )

            mockManyProductsFetch()
                .then(res => res.json())
                .then(res => {
                    wrapper.setState({
                        products: res,
                        isFetched: true
                    })

                    wrapper.update()

                    expect(wrapper.html()).toMatchSnapshot()

                    res.map(product => {
                        expect(wrapper.find(`a[href="/product/details/${product._id}"]`).length).toEqual(2)
                        expect(wrapper.find(`a[href="/product/like/${product._id}"]`).length).toEqual(1)
                        expect(wrapper.find(`a[href="/cart/add/${product._id}"]`).length).toEqual(1)
                    })
                })

        })

    })

    describe('admin functionalities tests', () => {
        it('should render loading gif for The [Admin] until the info is fetched', () => {
            const defaultUserValue = { isLogged: true, username: 'admin', role: 'Admin' }

            const wrapper = mount(
                <UserInfoProvider value={defaultUserValue}>
                    <AllProducts />
                </UserInfoProvider>
            )

            expect(wrapper.find('img[alt="loading-img"]').length).toEqual(1)
        })

        it('should render a div saying that there are no products for The [Admin]', () => {
            const defaultUserValue = { isLogged: false, username: 'admin4e', role: 'Admin' }

            const wrapper = mount(
                <UserInfoProvider value={defaultUserValue}>
                    <AllProducts service={mockNoProductsFetch} />
                </UserInfoProvider>
            )

            mockNoProductsFetch()

                .then(res => res.json())
                .then(res => {
                    wrapper.setState({
                        products: res,
                        isFetched: true
                    })

                    wrapper.update()

                    expect(wrapper.html()).toMatchSnapshot()

                    expect(wrapper.find('div.no-products-container').length).toEqual(1)
                    expect(wrapper.find('div.no-products-text-container').length).toEqual(1)

                    expect(wrapper.find('h1').length).toEqual(1)
                    expect(wrapper.find('h1').text()).toEqual('Our products are so desired that we could not predict we would run out of stock this soon..')

                    expect(wrapper.find('h2').length).toEqual(1)
                    expect(wrapper.find('h2').text()).toEqual('Sorry!')

                    expect(wrapper.find('div.no-products-img').length).toEqual(1)
                })
        })

        it('should render one product with links for [like][edit][delete] and [add to cart] below for The [Admin]', () => {
            const defaultUserValue = { isLogged: true, username: 'admin', role: 'Admin' }

            const wrapper = mount(
                <BrowserRouter>
                    <UserInfoProvider value={defaultUserValue}>
                        <AllProducts service={mockAProductFetch} />
                    </UserInfoProvider>
                </BrowserRouter>
            )

            mockAProductFetch()
                .then(res => res.json())
                .then(res => {

                    wrapper.setState({
                        products: res,
                        isFetched: true
                    })
                    wrapper.update()

                    expect(wrapper.html()).toMatchSnapshot()
                    expect(wrapper.find('a[href="/product/details/1"]').length).toEqual(2)

                    expect(wrapper.find('a[href="/product/like/1"]').length).toEqual(1)
                    expect(wrapper.find('a[href="/product/edit/1"]').length).toEqual(1)
                    expect(wrapper.find('a[href="/product/delete/1"]').length).toEqual(1)
                    expect(wrapper.find('a[href="/cart/add/1"]').length).toEqual(1)
                })
        })

        it('should render three products with links for [like][edit][delete] and [add to cart] below for The [Admin]', () => {
            const defaultUserValue = { isLogged: true, username: 'admin', role: 'Admin' }

            const wrapper = mount(
                <BrowserRouter>
                    <UserInfoProvider value={defaultUserValue}>
                        <AllProducts service={mockManyProductsFetch} />
                    </UserInfoProvider>
                </BrowserRouter>
            )

            mockManyProductsFetch()
                .then(res => res.json())
                .then(res => {
                    wrapper.setState({
                        products: res,
                        isFetched: true
                    })

                    wrapper.update()

                    expect(wrapper.html()).toMatchSnapshot()

                    res.map(product => {
                        expect(wrapper.find(`a[href="/product/details/${product._id}"]`).length).toEqual(2)
                        expect(wrapper.find(`a[href="/product/like/${product._id}"]`).length).toEqual(1)
                        expect(wrapper.find(`a[href="/product/edit/${product._id}"]`).length).toEqual(1)
                        expect(wrapper.find(`a[href="/product/delete/${product._id}"]`).length).toEqual(1)
                        expect(wrapper.find(`a[href="/cart/add/${product._id}"]`).length).toEqual(1)
                    })
                })
        })
    })
})