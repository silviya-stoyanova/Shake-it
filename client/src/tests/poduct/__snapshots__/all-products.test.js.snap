// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for the component AllProducts, a.k.a. Home page admin functionalities tests should render a div saying that there are no products for The [Admin] 1`] = `"<div class=\\"content-wrapper\\"><div class=\\"no-products-container\\"><div class=\\"no-products-text-container\\"><h1>Our products are so desired that we could not predict we would run out of stock this soon..</h1><h2>Sorry!</h2></div><div class=\\"no-products-img\\"></div></div></div>"`;

exports[`tests for the component AllProducts, a.k.a. Home page admin functionalities tests should render one product with links for [like][edit][delete] and [add to cart] below for The [Admin] 1`] = `"<div class=\\"content-wrapper\\"><div><label for=\\"search\\">Find your milkshake..</label><input name=\\"search\\" type=\\"text\\" id=\\"search\\"></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/1\\">Tova e title</a><a href=\\"/product/details/1\\"><img src=\\"data:image/png;base64, img\\" alt=\\"Tova e title\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 5<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">3 ♥</span><a class=\\"product-actions-btn\\" href=\\"/product/like/1\\">like</a><a class=\\"product-actions-btn\\" href=\\"/product/edit/1\\">edit</a><a class=\\"product-actions-btn\\" href=\\"/product/delete/1\\">delete</a><a class=\\"product-actions-btn add-to-cart-btn\\" href=\\"/cart/add/1\\">add to cart</a></div></div></div>"`;

exports[`tests for the component AllProducts, a.k.a. Home page admin functionalities tests should render three products with links for [like][edit][delete] and [add to cart] below for The [Admin] 1`] = `"<div class=\\"content-wrapper\\"><div><label for=\\"search\\">Find your milkshake..</label><input name=\\"search\\" type=\\"text\\" id=\\"search\\"></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/1\\">Tova e title</a><a href=\\"/product/details/1\\"><img src=\\"data:image/png;base64, img\\" alt=\\"Tova e title\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 5<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">3 ♥</span><a class=\\"product-actions-btn\\" href=\\"/product/like/1\\">like</a><a class=\\"product-actions-btn\\" href=\\"/product/edit/1\\">edit</a><a class=\\"product-actions-btn\\" href=\\"/product/delete/1\\">delete</a><a class=\\"product-actions-btn add-to-cart-btn\\" href=\\"/cart/add/1\\">add to cart</a></div></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/2\\">Tova e title 2</a><a href=\\"/product/details/2\\"><img src=\\"data:image/png;base64, img-2\\" alt=\\"Tova e title 2\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 6<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">2 ♥</span><a class=\\"product-actions-btn\\" href=\\"/product/like/2\\">like</a><a class=\\"product-actions-btn\\" href=\\"/product/edit/2\\">edit</a><a class=\\"product-actions-btn\\" href=\\"/product/delete/2\\">delete</a><a class=\\"product-actions-btn add-to-cart-btn\\" href=\\"/cart/add/2\\">add to cart</a></div></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/3\\">Tova e title 3</a><a href=\\"/product/details/3\\"><img src=\\"data:image/png;base64, img-3\\" alt=\\"Tova e title 3\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 3.8<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">1 ♥</span><a class=\\"product-actions-btn\\" href=\\"/product/like/3\\">like</a><a class=\\"product-actions-btn\\" href=\\"/product/edit/3\\">edit</a><a class=\\"product-actions-btn\\" href=\\"/product/delete/3\\">delete</a><a class=\\"product-actions-btn add-to-cart-btn\\" href=\\"/cart/add/3\\">add to cart</a></div></div></div>"`;

exports[`tests for the component AllProducts, a.k.a. Home page guest functionalities tests should render a div saying that there are no products for all [guests] 1`] = `"<div class=\\"content-wrapper\\"><div class=\\"no-products-container\\"><div class=\\"no-products-text-container\\"><h1>Our products are so desired that we could not predict we would run out of stock this soon..</h1><h2>Sorry!</h2></div><div class=\\"no-products-img\\"></div></div></div>"`;

exports[`tests for the component AllProducts, a.k.a. Home page guest functionalities tests should render one product without any links below it for all [guests] 1`] = `"<div class=\\"content-wrapper\\"><div><label for=\\"search\\">Find your milkshake..</label><input name=\\"search\\" type=\\"text\\" id=\\"search\\"></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/1\\">Tova e title</a><a href=\\"/product/details/1\\"><img src=\\"data:image/png;base64, img\\" alt=\\"Tova e title\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 5<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">3 ♥</span></div></div></div>"`;

exports[`tests for the component AllProducts, a.k.a. Home page guest functionalities tests should render three products without any links below it for all [guests] 1`] = `"<div class=\\"content-wrapper\\"><div><label for=\\"search\\">Find your milkshake..</label><input name=\\"search\\" type=\\"text\\" id=\\"search\\"></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/1\\">Tova e title</a><a href=\\"/product/details/1\\"><img src=\\"data:image/png;base64, img\\" alt=\\"Tova e title\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 5<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">3 ♥</span></div></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/2\\">Tova e title 2</a><a href=\\"/product/details/2\\"><img src=\\"data:image/png;base64, img-2\\" alt=\\"Tova e title 2\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 6<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">2 ♥</span></div></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/3\\">Tova e title 3</a><a href=\\"/product/details/3\\"><img src=\\"data:image/png;base64, img-3\\" alt=\\"Tova e title 3\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 3.8<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">1 ♥</span></div></div></div>"`;

exports[`tests for the component AllProducts, a.k.a. Home page user functionalities tests should render a div saying that there are no products for all [users] 1`] = `"<div class=\\"content-wrapper\\"><div class=\\"no-products-container\\"><div class=\\"no-products-text-container\\"><h1>Our products are so desired that we could not predict we would run out of stock this soon..</h1><h2>Sorry!</h2></div><div class=\\"no-products-img\\"></div></div></div>"`;

exports[`tests for the component AllProducts, a.k.a. Home page user functionalities tests should render one product with links for [like] and [add to cart] below for all [users] 1`] = `"<div class=\\"content-wrapper\\"><div><label for=\\"search\\">Find your milkshake..</label><input name=\\"search\\" type=\\"text\\" id=\\"search\\"></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/1\\">Tova e title</a><a href=\\"/product/details/1\\"><img src=\\"data:image/png;base64, img\\" alt=\\"Tova e title\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 5<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">3 ♥</span><a class=\\"product-actions-btn\\" href=\\"/product/like/1\\">like</a><a class=\\"product-actions-btn add-to-cart-btn\\" href=\\"/cart/add/1\\">add to cart</a></div></div></div>"`;

exports[`tests for the component AllProducts, a.k.a. Home page user functionalities tests should render three products with links for [like] and [add to cart] below for all [users] 1`] = `"<div class=\\"content-wrapper\\"><div><label for=\\"search\\">Find your milkshake..</label><input name=\\"search\\" type=\\"text\\" id=\\"search\\"></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/1\\">Tova e title</a><a href=\\"/product/details/1\\"><img src=\\"data:image/png;base64, img\\" alt=\\"Tova e title\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 5<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">3 ♥</span><a class=\\"product-actions-btn\\" href=\\"/product/like/1\\">like</a><a class=\\"product-actions-btn add-to-cart-btn\\" href=\\"/cart/add/1\\">add to cart</a></div></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/2\\">Tova e title 2</a><a href=\\"/product/details/2\\"><img src=\\"data:image/png;base64, img-2\\" alt=\\"Tova e title 2\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 6<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">2 ♥</span><a class=\\"product-actions-btn\\" href=\\"/product/like/2\\">like</a><a class=\\"product-actions-btn add-to-cart-btn\\" href=\\"/cart/add/2\\">add to cart</a></div></div><div class=\\"product-wrapper\\"><a class=\\"product-title\\" href=\\"/product/details/3\\">Tova e title 3</a><a href=\\"/product/details/3\\"><img src=\\"data:image/png;base64, img-3\\" alt=\\"Tova e title 3\\" class=\\"product-img\\"></a><div class=\\"product-actions\\"><span class=\\"product-price\\">Price: 3.8<span class=\\"price-sign\\">$</span></span><span class=\\"product-likes\\">1 ♥</span><a class=\\"product-actions-btn\\" href=\\"/product/like/3\\">like</a><a class=\\"product-actions-btn add-to-cart-btn\\" href=\\"/cart/add/3\\">add to cart</a></div></div></div>"`;
